nomt/src/beatree/mod.rs

```diff
@@ -288,10 +284,14 @@ pub fn create(db_dir: impl AsRef<Path>) -> anyhow::Result<()> {
 /// should be discarded.
 // TODO: error handling is coming in a follow up.
 pub struct SyncController {
-    tree: Tree,
-    tp: ThreadPool,
-    sync_data: Arc<Mutex<Option<SyncData>>>,
-    bbn_index: Arc<Mutex<Option<Index>>>,
+    inner: Arc<SharedSyncController>,
```

Comment:
Why an `Arc<SharedSyncController>` if the documentation says "This controller is designed to be used from a single thread"? Also, how can an `ArcMutexGuard` be shared? Shouldn't it be supposed to be used just by who called `tre.sync()`

Reply:
As per rustdoc, `SyncController` is designed to be called from a single thread, namely the thread that executes the store sync. The arc is used because under the hood the controller spawns some background operations.

Now, regarding the ArcMutexGuard behind an Arc... this is actually a good question and I am not sure how this works and why it even compiles. I have serious questions to the person who wrote that code...

Reply:
OK, had another look, well, and it is alright.

So why is arc mutex guard sus behind an arc sus? It seemed sus because multiple threads owning different handles to a single `struct Sync` instance. But even if that's the case, you can only ever get a shared reference to the guard protected data, which means:

1. modification of things without interior mutability is trivially disallowed,
2. things that allow interior mutability but are not Sync (think of Rc), are not allowed in the code, because that would make `struct Sync` not possible to store in a Mutex.
3. things that allow interior mutability but are Sync are OK, e.g. mutex.

so all in all, this is a non-issue.

Reply:
>The arc is used because under the hood the controller spawns some background operations.

oh, yes, totally, I see now!

>so all in all, this is a non-issue.

Yes, thanks for the broader explanation. Technically, it is a non-issue and I also understand the reason behind it now